src/wallet/api/wallet.cpp

```diff
@@ -552,28 +560,34 @@ bool WalletImpl::recoverFromKeysWithPassword(const std::string &path,
             return false;
         }
     }
-    if (!crypto::secret_key_to_public_key(viewkey, pkey)) {
-        m_errorString = tr("failed to verify secret view key");
-        m_status = Status_Error;
-        return false;
+    if(has_viewkey) {
+       if (!crypto::secret_key_to_public_key(viewkey, pkey)) {
+           m_errorString = tr("failed to verify secret view key");
+           m_status = Status_Error;
+           return false;
+       }
+       if (info.address.m_view_public_key != pkey) {
+           m_errorString = tr("view key does not match address");
+           m_status = Status_Error;
+           return false;
+       }
     }
-    if (info.address.m_view_public_key != pkey) {
-        m_errorString = tr("view key does not match address");
-        m_status = Status_Error;
-        return false;
-    }
-
     try
     {
-        if (has_spendkey) {
+        if (has_spendkey && has_viewkey) {
             m_wallet->generate(path, password, info.address, spendkey, viewkey);
             setSeedLanguage(language);
```

Comment:
I think the seed language doesn't matter in this case since the wallet is not deterministic (i.e. viewkey isn't derived from spendkey).

Reply:
I didn't touch that. It would set the seed language regardless of whether or not it was a deterministic wallet. Do you want me to remove it?

Reply:
+1 Done